pallets/committee/src/lib.rs

```diff
@@ -462,10 +462,8 @@ pub mod pallet {
 				}
 			})?;
 
-			VotingEligibility::<T>::insert(
-				&constituent,
-				frame_system::Pallet::<T>::block_number() + T::VotingPeriod::get(),
-			);
+			let block_numer = frame_system::Pallet::<T>::block_number();
+			VotingEligibility::<T>::insert(&constituent, block_numer + Self::get_next_voting_period_end(&block_numer)?);
```

Comment:
Dispatch the eligibility with `get_next_voting_period_end` since `Proposals`' end are generated by this function

BTW, not sure what is `ProposalSubmission` for, since different proposals have different voting period, this config terms to be no use

Reply:
submitting a proposal and submitting a vote are two different periods.

Reply:
If we keep this `SubmissionPeriod` we should add a check into `propose` call to ensure that people are submitting a proposal in SubmissionPeriod ? Currently there are no checks in `propose` call with `SubmissionPeriod`